posts/2024-11-19-24.0.0.12-beta.adoc

```diff
@@ -0,0 +1,240 @@
+
+
+
+=== Default validation
+
+You can configure the default level of JAXB validation of inbound SOAP Messages with the `enableDefaultValidation=true` attribute. Default validation is much more efficient than XML schema validation, so enabling it provides basic message validation with lower overhead. Disabling it lets you ignore various unmarshalling errors for problematic messages. Default validation is enabled by default for web services, but disabled for web service clients.
+
+==== Global default validation
+
+The following example shows how to enable default validation globally for web services in your Open Liberty runtime:
+
+[source,xml]
+----
+<webService enableDefaultValidation="true"/>
+----
+
+Default validation is enabled by default for web service clients. To disable default validation globally for web service clients, set the `enableDefaultValidation="false"` attribute on the `webServiceClient` element.
+
+[source,xml]
+----
+<webServiceClient enableDefaultValidation="false"/>
+----
+
+==== Targeted default validation
+
+The following example shows how to enable default validation for a specific web service:
+
+[source,xml]
+----
+<webService  portName="SayHelloService" enableDefaultValidation="true"/>
+----
+
+Default validation is enabled by default for web service clients. To disable default validation for a specific web service client, set the `enableDefaultValidation="false"` attribute on the `webServiceClient` element and use the `serviceName` attribute to specify the client service.
+
+[source,xml]
+----
+<webServiceClient serviceName="<client service name>"  enableDefaultValidation="false" />
+----
+
+=== Ignore unexpected elements
+
+The new Open Liberty `ignoreUnexpectedElements` attribute provides the same function. Inbound SOAP messages often contain extra elements in the SOAP body when a web service is updated but the client is not. When a message contains an unknown element, Open Liberty throws a `UnmarshallingException: Unknown Element`. By enabling `ignoreUnkownEllements`, you can keep validation enabled while ignoring unknown elements.
+
+==== Global configuration
+
+The following example shows how to ignore unexpected elements globally for web services on your Open Liberty runtime:
+
+[source,xml]
+----
+<webService  ignoreUnexpectedElements="true"/>
+----
+
+To ignore unexpected elements globally for web service clients, set the `ignoreUnexpectedElements` attribute on the `webServiceClient` element.
+
+==== Targeted configuration
+
+The following example shows how to ignore unexpected elements for a specific web service:
+
+[source,xml]
+----
+<webService  portName="SayHelloService" ignoreUnexpectedElements="true"/>
+----
+
+To ignore unexpected elements for a specific web service client, set the same attribute on the `webServiceClient`element and use the `serviceName` attribute to specify the client service.
```

Comment:
```suggestion
To ignore unexpected elements for a specific web service client, set the same attribute on the `webServiceClient` element and use the `serviceName` attribute to specify the client service.
```

Reply:
addressed in https://github.com/OpenLiberty/blogs/pull/4151